proto/prysm/v1alpha1/attestation/attestation_utils.go

```diff
@@ -55,32 +55,72 @@ func ConvertToIndexed(ctx context.Context, attestation interfaces.Attestation, c
-//	                       data: AttestationData,
-//	                       bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]) -> Set[ValidatorIndex]:
-//	 """
-//	 Return the set of attesting indices corresponding to ``data`` and ``bits``.
-//	 """
-//	 committee = get_beacon_committee(state, data.slot, data.index)
-//	 return set(index for i, index in enumerate(committee) if bits[i])
-func AttestingIndices(bf bitfield.Bitfield, committee []primitives.ValidatorIndex) ([]uint64, error) {
-	if bf.Len() != uint64(len(committee)) {
-		return nil, fmt.Errorf("bitfield length %d is not equal to committee length %d", bf.Len(), len(committee))
-	}
-	indices := make([]uint64, 0, bf.Count())
-	p := bf.BitIndices()
-	for _, idx := range p {
-		if idx < len(committee) {
-			indices = append(indices, uint64(committee[idx]))
+//	def get_attesting_indices(state: BeaconState, attestation: Attestation) -> Set[ValidatorIndex]:
+//	    """
+//	    Return the set of attesting indices corresponding to ``aggregation_bits`` and ``committee_bits``.
+//	    """
+//	    output: Set[ValidatorIndex] = set()
+//	    committee_indices = get_committee_indices(attestation.committee_bits)
+//	    committee_offset = 0
+//	    for index in committee_indices:
+//	        committee = get_beacon_committee(state, attestation.data.slot, index)
+//	        committee_attesters = set(
+//	        index for i, index in enumerate(committee) if attestation.aggregation_bits[committee_offset + i])
+//	        output = output.union(committee_attesters)
+//
+//	        committee_offset += len(committee)
+//
+//	    return output
+func AttestingIndices(att interfaces.Attestation, committees [][]primitives.ValidatorIndex) ([]uint64, error) {
+	if len(committees) == 0 {
+		return []uint64{}, nil
+	}
+
+	aggBits := att.GetAggregationBits()
+
+	if att.Version() < version.Electra {
+		committee := committees[0]
+		if aggBits.Len() != uint64(len(committee)) {
+			return nil, fmt.Errorf("bitfield length %d is not equal to committee length %d", aggBits.Len(), len(committee))
+		}
+		indices := make([]uint64, 0, aggBits.Count())
+		p := aggBits.BitIndices()
+		for _, idx := range p {
+			if idx < len(committee) {
+				indices = append(indices, uint64(committee[idx]))
+			}
 		}
+		return indices, nil
 	}
-	return indices, nil
+
+	committeesLen := 0
+	for _, c := range committees {
+		committeesLen += len(c)
+	}
+	if aggBits.Len() != uint64(committeesLen) {
+		return nil, fmt.Errorf("bitfield length %d is not equal to committee length %d", aggBits.Len(), committeesLen)
+	}
+
+	attesters := make([]uint64, 0, aggBits.Count())
+	committeeOffset := 0
+	for _, c := range committees {
+		committeeAttesters := make([]uint64, 0, len(c))
+		for i, vi := range c {
+			if aggBits.BitAt(uint64(committeeOffset + i)) {
+				committeeAttesters = append(committeeAttesters, uint64(vi))
+			}
+		}
+		attesters = append(attesters, committeeAttesters...)
+		committeeOffset += len(c)
+	}
+
+	return attesters, nil
```

Comment:
The spec expects that the resulting attesters is a distinct set with no duplicates. Could you rewrite this to ensure there are no duplicate validator indices? 

This test case fails:

```diff
diff --git a/proto/prysm/v1alpha1/attestation/attestation_utils_test.go b/proto/prysm/v1alpha1/attestation/attestation_utils_test.go
index 2ebf046a02..2b032bda1a 100644
--- a/proto/prysm/v1alpha1/attestation/attestation_utils_test.go
+++ b/proto/prysm/v1alpha1/attestation/attestation_utils_test.go
@@ -74,6 +74,14 @@ func TestAttestingIndices(t *testing.T) {
                        },
                        err: "bitfield length 5 is not equal to committee length 4",
                },
+         {
+                       name: "Electra - No duplicates",
+                       args: args{
+                               att:        &eth.AttestationElectra{AggregationBits: bitfield.Bitlist{0b11111}},
+                               committees: [][]primitives.ValidatorIndex{{0, 1}, {0, 1}},
+                       },
+                       want: []uint64{0, 1},
+               },
        }
        for _, tt := range tests {
                t.Run(tt.name, func(t *testing.T) {
```

Reply:
oh, great find